components/renku_data_services/session/api.spec.yaml

```diff
@@ -0,0 +1,267 @@
+              schema:
+                $ref: "#/components/schemas/Session"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - sessions
+  /sessions/{session_id}:
+    get:
+      summary: Get a session
+      parameters:
+        - in: path
+          name: session_id
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: The session
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Session"
+        "404":
+          description: The session does not exist
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - sessions
+    patch:
+      summary: Update specific fields of an existing session
+      parameters:
+        - in: path
+          name: session_id
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SessionPatch"
+      responses:
+        "200":
+          description: The patched session
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Session"
+        "404":
+          description: The session does not exist
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - sessions
+    delete:
+      summary: Remove a session
+      parameters:
+        - in: path
+          name: session_id
+          required: true
+          schema:
+            type: string
+      responses:
+        "204":
+          description: The session was removed or did not exist in the first place
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - sessions
+components:
+  schemas:
+    SessionsList:
+      description: A list of Renku sessions
+      type: array
+      items:
+        $ref: "#/components/schemas/Session"
+      minItems: 0
+    Session:
+      description: A Renku native session definition and metadata
+      type: object
+      properties:
+        # TODO: Do we need a slug for sessions!?
+        id:
+          $ref: "#/components/schemas/Ulid"
+        name:
+          $ref: "#/components/schemas/Name"
+        creation_date:
+          $ref: "#/components/schemas/CreationDate"
+        description:
+          $ref: "#/components/schemas/Description"
+        environment_id:
+          $ref: "#/components/schemas/EnvironmentId"
+        status:
+          $ref: "#/components/schemas/Status"
+        tags:
+          $ref: "#/components/schemas/TagsList"
```

Comment:
I don't think we have anything yet in the pitch about tags. We would have to understand where to put them between sessions and environments, so I'm not sure we can use them now.

Reply:
I'll remove them for now.